### PR TITLE
Week 7 Lab notes

### PR DESCRIPTION
**Pull request by Heimine**
We discussed some applications of Autoencoders and talked about why we want to use them. Then we talked about different architectures of Autoencoders (under or over complete hidden layer), how to avoid overfitting issues and loss functions we should use. Finally we implemented a standard Autoencoder and a denoising Autoencoder.

### CODE REVIEW

**Comment by Atcold** [comment]
A few notes.
Remove fig 20.
Cut 3/4 of fig 21.
About the code, people *do* have access to the notebook, so you shouldn't copy and paste *all* the code.
Only the relevant bits, necessary to make the point of the given class.
Also, we don't care about `print`, `display`, and so on. These are auxiliary functions that are only cluttering the reading. Remove `to_img` and `display_images` definitions. These are absolutely *not* relevant.
